### PR TITLE
Add more specific instruction on applying plugins

### PR DESCRIPTION
**Pull request by MarcinAman**

### CODE REVIEW

File: docs/src/doc/docs/user_guide/gradle/usage.md [hunk 1]
```diff
@@ -226,7 +226,19 @@ tasks.withType<DokkaTask>().configureEach {
...
+| :------------- | :------------------------------------------------- | :----------- |
+| base           | None, applied by default                           | dokkaHtml    |
+| gfm            | None, applied by default                           | dokkaGfm     |
+| jekyll         | None, applied by default                           | dokkaJekyll  |
+| javadoc        | None, applied by default                           | dokkaJavadoc |
+| kotlin-as-java | `org.jetbrains.dokka:kotlin-as-java-plugin:1.4.10` | None         |
+
+By default, you will get all the plugins except `kotlin-as-java`. There is no need to apply them manually.
+
+In order to apply a plugin add it in `dependencies` block:
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/gradle/usage.md:241]:
```suggestion
In order to apply a plugin add it in the `dependencies` block:
```

File: docs/src/doc/docs/user_guide/gradle/usage.md [hunk 2]
```diff
@@ -253,6 +265,8 @@ To generate the documentation, use the appropriate `dokka${format}` Gradle task:
 ./gradlew dokkaHtml
 ```
 
+Keep in mind, that plugins developed by the community may not follow same convention. Always consult plugin's creator documentation.
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/gradle/usage.md:268]:
```suggestion
!!! important
    Keep in mind that plugins developed by the community may not follow the same convention. Always consult the plugin's documentation.
```

File: docs/src/doc/docs/user_guide/plugins-specific/base-specific/frontend.md [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
  
 Every file provided in those values will be applied to **every** page.
 
 Dokka uses 3 stylesheets:
-* `style.css` - main css file responsible for styling the page
-* `jetbrains-mono.css` - fonts used across dokka
-* `logo-styles.css` - logo styling
+
+ * `style.css` - main css file responsible for styling the page
+ * `jetbrains-mono.css` - fonts used across dokka
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/plugins-specific/base-specific/frontend.md:18]:
```suggestion
 * `jetbrains-mono.css` - fonts used across Dokka
```

File: docs/src/doc/docs/user_guide/plugins-specific/index.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Documentation for specific plugin
+
+Currently, dokka offers 4 plugins:
+
+ * [base plugin with HTML format](base-specific/frontend.md)
+ * GFM plugin
+ * Jekyll plugin
+ * Kotlin-as-Java plugin
+
+Base, GFM and Jekyll plugins are applied by default. Rest of those needs to be applied manually.
```
Comment (kamildoleglo) [inline docs/src/doc/docs/user_guide/plugins-specific/index.md:10]:
```suggestion
Base, GFM, and Jekyll plugins are applied by default. The rest of those need to be applied manually. 
```

**Review by kamildoleglo — CHANGES_REQUESTED** [review]

File: docs/src/doc/docs/user_guide/gradle/usage.md [hunk 5]
```diff
@@ -226,7 +226,19 @@ tasks.withType<DokkaTask>().configureEach {
     If you want to share the configuration between source sets, you can use Gradle's `configureEach`
 
 ## Applying plugins
-Dokka plugin creates Gradle configuration for each output format in the form of `dokka${format}Plugin`:
+Dokka allows users to create and use custom plugins. Out of the box it comes with:
+
+| Plugin name    | Dependency                                         | Added tasks  |
```
[the remark this answers is not in the document]
Comment (MarcinAman) [reply docs/src/doc/docs/user_guide/gradle/usage.md:231]: This is a dependency that would need to be added to the build in order to use this feature not a plugins hierarchy

[the remark this answers is not in the document]
Comment (kamildoleglo) [reply docs/src/doc/docs/user_guide/gradle/usage.md:231]: Yes, but `Dependency` looks like its the only dependency of the plugin, which is not true

[the remark this answers is not in the document]
Comment (MarcinAman) [reply docs/src/doc/docs/user_guide/gradle/usage.md:231]: applied with

**Review by kamildoleglo — APPROVED** [review]

**Comment by IgnatBeresnev** [comment]
The documentation was re-written and is now published to kotlinlang.org. There's a section on Dokka plugins and how to apply them: https://kotlinlang.org/docs/dokka-plugins.html

Closing the PR as implemented / outdated

**Pull request closed without merging**